pkg/dbmate/db.go

```diff
@@ -441,6 +453,10 @@ func (db *DB) FindMigrations() ([]Migration, error) {
 				migration.Applied = true
 			}
 
+			if db.Strict && !migration.Applied && migration.Version < latestAppliedMigration {
```

Comment:
This is not the right place for this check to live. `FindMigrations()` should not fail regardless of what migrations have or haven't been applied.

The check should be implemented in the `Migrate()` function

Reply:
Without an additional db query the check needs access to a list of applied migrations to find the applied migration with the highest version number. `FindMigrations()` already queries those to check if a migration is already applied. There are several ways to do the check outside of `FindMigrations()` but I did not want to make too many unrelated changes which might be in conflict with other PRs, e.g. changing the return values of `FindMigrations()` would be in conflict with PR #438.

Reply:
I put the strict version check in `Migrate()` and applied some small changes. `FindMigrations()` already returns a sorted list of migrations so we only need to check the first pendig migration with the highest version number from all applied migrations.